lesson_19/page_objects/login_page.py

```diff
@@ -0,0 +1,60 @@
+from selenium.webdriver.common.by import By
+
+from .password_recovery_page import PasswordRecovery
+from ..page_objects.base_page import BasePage
+from ..page_objects.register_page import RegisterPage
+
+
+class LoginPage(BasePage):
+    def __init__(self, driver):
+        super().__init__(driver)
+
+    __email_field = (By.XPATH, "//input[@id='Email']")
+    __password_field = (By.XPATH, "//input[@id='Password']")
+    __log_in_button = (By.XPATH, "//input[@class='button-1 login-button']")
+    __register_button = (By.XPATH, "//div[@class='buttons']//input[@class='button-1 register-button']")
+    __remember_checkbox = (By.XPATH, "//input[@id='RememberMe']")
+    __forgot_password = (By.XPATH, "//span[@class='forgot-password']//a")
+
+    __login_error_status = (By.XPATH, "//div[@class='validation-summary-errors']//span")
+    __login_error_cause = (By.XPATH, "//div[@class='validation-summary-errors']//ul//li")
+
+    def set_email(self, email: str):
+        email_field = self._wait_element(self.__email_field, type_of='clickable')
+        email_field.clear()
+        email_field.send_keys(email)
```

Comment:
you have a base send_keys function. Why don't you use it here&

Reply:
It looks like I forgot about it. Fixed